crates/core/src/marzano_binding.rs

```diff
@@ -52,14 +53,35 @@ impl PartialEq for MarzanoBinding<'_> {
     }
 }
 
-fn pad_snippet(padding: &str, snippet: &str) -> String {
-    // Write first snippet line as is, without extra padding
+fn get_skip_padding_ranges_for_snippet(
+    lang: &impl Language,
+    snippet: &str,
+) -> Result<Vec<CodeRange>> {
+    let mut parser = TSParser::new()?;
+    parser.set_language(lang.get_ts_language())?;
+    let tree = parser
+        .parse(snippet, None)?
+        .ok_or(anyhow!("failed to parse snippet"))?;
+    let root = tree.root_node();
+    let node = NodeWithSource::new(root, snippet);
+    Ok(get_skip_padding_ranges(&node, lang))
+}
+
+pub(crate) fn pad_snippet(padding: &str, snippet: &str, lang: &impl Language) -> Result<String> {
     let mut lines = snippet.split('\n');
     let mut result = lines.next().unwrap_or_default().to_string();
 
     // Add the rest of lines in the snippet with padding
-    lines.for_each(|line| result.push_str(&format!("\n{}{}", &padding, line)));
-    result
+    let skip_ranges = get_skip_padding_ranges_for_snippet(lang, snippet)?;
```

Comment:
This doesn't look optimal, why are we needing to re-parse here?

Reply:
we're only parsing a snippet, it's to check if it's a string, if it's a string we don't want to pad it

Reply:
Sure, but we don't actually know/guarantee RHS snippets are parseable. We had the original binding at some point, we should be able to get the sort from there.

Reply:
The original binding if there was one is lost at this point as a snippet is made up of a patchwork of bindings that have been linearized by this point. It doesn't matter if we the snippet is parsed without error, all we are looking for are strings if we get error nodes instead of strings then the worst case is we padded when we should have, but I find it unlikely that we will have a node with a string that doesn't parse as a string node. Consider the rhs `"""$content"""` content wasn't a string, but we turned it into a string on the rhs, so there's no original binding with the information we need.

Reply:
Ok, I still don't love this approach but it's probably only a minor performance hit.